src/aics-image-viewer/components/TfEditor/index.tsx

```diff
@@ -0,0 +1,818 @@
+  private keydown(_e: KeyboardEvent) {
+    if (!this.selected) {
+      return;
+    }
+    if (d3.event.keyCode === 46) {
+      // delete
+      const i = this.props.controlPoints.indexOf(this.selected);
+      let newControlPoints = [...this.props.controlPoints];
+      newControlPoints.splice(i, 1);
+      this.selected = newControlPoints.length > 0 ? newControlPoints[i > 0 ? i - 1 : 0] : null;
+      this.props.updateChannelLutControlPoints(newControlPoints);
+    }
+  }
+
+  // TODO unused
+  private export() {
+    const jsonContent = JSON.stringify(this.props.controlPoints);
+    const a = document.createElement("a");
+    const blob = new Blob([jsonContent], {
+      type: "octet/stream",
+    });
+    const url = window.URL.createObjectURL(blob);
+    a.href = url;
+    a.download = "transferFunction.json";
+    a.click();
+    window.URL.revokeObjectURL(url);
+  }
+
+  updateControlPointsWithoutColor(ptsWithoutColor: ControlPoint[]) {
+    const pts = ptsWithoutColor.map((pt) => ({
+      ...pt,
+      color: TFEDITOR_DEFAULT_COLOR,
+    }));
+    this.updateControlPoints(pts);
+  }
+
+  updateControlPoints(pts: ControlPoint[]) {
+    // TODO do I need to copy the pts here?
+    this.selected = pts[0];
+    this.props.updateChannelLutControlPoints(pts);
+  }
+
+  private autoXF() {
+    const { channelData } = this.props;
+
+    const lutObj = channelData.histogram.lutGenerator_auto();
+    this.updateControlPointsWithoutColor(lutObj.controlPoints);
+  }
+
+  private auto2XF() {
+    const { channelData } = this.props;
+
+    const lutObj = channelData.histogram.lutGenerator_auto2();
+    this.updateControlPointsWithoutColor(lutObj.controlPoints);
+  }
+
+  private auto98XF() {
+    const { channelData } = this.props;
+
+    const lutObj = channelData.histogram.lutGenerator_percentiles(LUT_MIN_PERCENTILE, LUT_MAX_PERCENTILE);
+    this.updateControlPointsWithoutColor(lutObj.controlPoints);
+  }
+
+  private bestFitXF() {
+    const { channelData } = this.props;
+
+    const lutObj = channelData.histogram.lutGenerator_bestFit();
+    this.updateControlPointsWithoutColor(lutObj.controlPoints);
+  }
+
+  private handleColorizeCheckbox(e: CheckboxChangeEvent) {
+    this.props.updateColorizeMode(e.target.checked);
+  }
+
+  private handleColorizeAlpha(values: number[]) {
+    this.props.updateColorizeAlpha(values[0]);
+  }
+
+  private resetXF() {
+    const { channelData } = this.props;
+
+    const lutObj = channelData.histogram.lutGenerator_fullRange();
+    this.updateControlPointsWithoutColor(lutObj.controlPoints);
+  }
+
+  /////// Public API functions ///////
+
+  /** Set the pixel data we are manipulating */
+  setData() {
+    if (!this.props.channelData) {
+      throw new Error("Transfer Function Editor setData called with no channel data.");
+    }
+    this.updateScales();
+    this.updateAxis();
+  }
+
+  /////// Polymer lifecycle callbacks /////////////
```

Comment:
this comment is obsolete and should be removed. the original implementation of this component was as a polymer component but it has long since been ported into react